File: backup/Twitch/app/src/main/java/com/exact/twitch/ui/videos/ChannelVideosFragment.java
package com.exact.twitch.ui.videos;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import android.view.View;

import com.exact.twitch.R;
import com.exact.twitch.ui.fragment.RadioButtonBottomSheetDialogFragment;
import com.exact.twitch.util.FragmentUtils;

public class ChannelVideosFragment extends BaseVideosFragment implements RadioButtonBottomSheetDialogFragment.OnOptionSelectedListener {

    private int[] sortOptions = new int[] { R.string.upload_date, R.string.view_count };

    private Object channelId;

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        if (isFragmentVisible()) {
            int defaultSortItem = sortOptions[0];
            if (getViewModel().getSelectedOrderItem() == null) {
                getViewModel().setSelectedOrderItem(defaultSortItem);
            }
            if (getViewModel().getCurrentOrderText().getValue() == null) {
                getViewModel().getCurrentOrderText().setValue(getString(defaultSortItem));
            }
            if (getViewModel().getSort() == null) {
                getViewModel().setSort("time");
            }
            getViewModel().getCurrentOrderText().observe(this, currentSortOption::setText);
        }
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        if (getArguments() != null) {
            channelId = getArguments().get("channelId");
        }
        sortByContainer.setOnClickListener(v -> FragmentUtils.INSTANCE.showRadioButtonDialogFragment(
                requireActivity(),
                getChildFragmentManager(),
                sortOptions,
                getViewModel().getSelectedOrderItem())
        );
    }

    @Override
    public void onSelect(int optionId, CharSequence optionText) {
        if (getViewModel().getSelectedOrderItem() != optionId) {
            getViewModel().setSort(optionId == R.string.upload_date ? "time" : "views");
            getViewModel().setSelectedOrderItem(optionId);
            getViewModel().getCurrentOrderText().postValue(optionText);
            loadData(true);
        }
    }

    @Override
    public void loadData(boolean override) {
        getViewModel().loadChannelVideos(channelId, null, getViewModel().getSort(), override);
    }

    @Override
    protected void initData() {
        loadData(false);
    }
}

File: app/src/main/java/com/exact/xtra/ui/videos/ChannelVideosFragment.kt
package com.exact.xtra.ui.videos

import android.content.Context
import android.os.Bundle
import android.view.View

import com.exact.xtra.R
import com.exact.xtra.ui.fragment.RadioButtonDialogFragment
import com.exact.xtra.util.C
import com.exact.xtra.util.FragmentUtils
import kotlinx.android.synthetic.main.fragment_videos.*

class ChannelVideosFragment : BaseVideosFragment(), RadioButtonDialogFragment.OnSortOptionChanged {

    private companion object {
        val sortOptions = listOf(R.string.upload_date, R.string.view_count)
        const val DEFAULT_INDEX = 0
        const val TAG = "ChannelVideos"
    }

    private lateinit var channelId: Any

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        channelId = arguments?.get("channelId")!!
        sortBar.setOnClickListener { FragmentUtils.showRadioButtonDialogFragment(requireActivity(), childFragmentManager, sortOptions, DEFAULT_INDEX, TAG) }
    }

    override fun initializeViewModel() {
        val index = requireActivity().getSharedPreferences(C.USER_PREFS, Context.MODE_PRIVATE).getInt(TAG, DEFAULT_INDEX)
        viewModel.sortText.postValue(getString(sortOptions[index]))
        viewModel.sort = if (index == 0) Sort.TIME else Sort.VIEWS
    }

    override fun loadData(override: Boolean) {
        viewModel.loadChannelVideos(channelId = channelId, reload = override)
    }

    override fun onChange(index: Int, text: CharSequence, tag: Int?) { //TODO move this to viewmodel and update there
        viewModel.sort = if (tag == R.string.upload_date) Sort.TIME else Sort.VIEWS
        viewModel.sortText.postValue(text)
        loadData(true)
    }
}
